### ISSUE TITLE
Running in worker thread throws error "Module did not self-register"

### ISSUE THREAD

**Issue by jpdewing** [marudor/libxmljs2#173]
When using this package in a node worker thread it throws the error "Module did not self-register".  

Reading up on this error I found some comments in the nodejs github issues log : [https://github.com/nodejs/node/issues/21783#issuecomment-429637117](https://github.com/nodejs/node/issues/21783#issuecomment-429637117).  And more information about the c++ context aware addons can be found here [https://github.com/nodejs/node/blob/master/doc/api/addons.md#context-aware-addons](https://github.com/nodejs/node/blob/master/doc/api/addons.md#context-aware-addons).

I tried the fix mentioned in this [https://stackoverflow.com/questions/68598248/is-there-a-way-to-run-native-code-modules-using-worker-threads](https://stackoverflow.com/questions/68598248/is-there-a-way-to-run-native-code-modules-using-worker-threads) post, changing line 250 in src/libxmljs.cc from

NODE_MODULE(xmljs, init)
to 
NAN_MODULE_WORKER_ENABLED(xmljs, init)

This does fix the problem but to be honest I don't have the expertise to understand the implications of the change.  Could one of the maintainers who has a better understanding of the library look at the possibility of either introducing this change or investigating another way to make it context aware.

**Comment by jpdewing** [comment]
No, we ended up restructuring our code so that this library is not loaded by worker threads and is only used in the main thread.  This was possible because it was other logic that we wanted to offload into the workers.

**Comment by vforsh** [comment]
I see. Unfortunately it's not an option for us since the bottleneck is libxmljs methods calls.
